lib/runners/objc.js

```diff
@@ -1,37 +1,40 @@
 var shovel = require('../shovel'),
     util = require('../util'),
+    temp = require('temp'),
     path = require('path'),
-    temp = require('temp');
+    exec = require('child_process').exec;
+
+function compile(args, cb) {
+    args.unshift('gcc', '`gnustep-config --objc-flags`');
+    exec(args.join(' '), cb);
+}
+
 
 module.exports.run = function run(opts, cb) {
-    // Sandbox expects files to have this directory as an ancestor
-    var objcDir = '/tmp/objc';
-    util.mkdirParentSync(objcDir);
     temp.track();
-    var dir = temp.mkdirSync({dir: objcDir}),
-	      objcSandbox = path.resolve('frameworks', 'osx', 'objc.sb');
+    var dir = temp.mkdirSync('objc');
+    var args = [];
 
     shovel.start(opts, cb, {
         solutionOnly: function (runCode, fail) {
-            var solutionFile = util.codeWriteSync('objc', opts.code, dir, 'main.m'),
-                command = 'clang -lobjc -framework Foundation -o main ' + solutionFile;
-
-            if ("setup" in opts) {
-                util.codeWriteSync('objcHeader', opts.setupHeader, dir, 'setup.h');
-                command = command + ' ' + util.codeWriteSync('objc', opts.setup, dir, 'setup.m');
-            }
-
-            util.exec(command, {'cwd': dir}, function (error) {
-                if (error) return fail(error);
-                runCode({
-                    'name': 'sandbox-exec',
-                    'args': ['-f', objcSandbox, './main'],
-                    'options': {'cwd': dir}
-                });
+            var executable = path.join(dir, 'solution');
+
+            opts.solution =
+                '#import <Foundation/Foundation.h>\n' +
+                'int main (int argc, const char * argv[]) {\n' +
+                opts.solution + '\n' +
+                'return 0;\n' +
+                '}';
+
+            var solutionFile = util.codeWriteSync('objc', opts.solution, dir, 'solution.m');
+
+            args = [solutionFile, '-lgnustep-base', '-lobjc', '-o', executable];
+
+            compile(args, function (error, stdout, stderr) {
+                if (error) return fail(error, stdout, stderr);
+                opts.publish('stdout', stdout);
+                runCode({'name': executable, 'args': []});
             });
-        },
-        testIntegration: function (runCode, fail) {
-            require('./xcode').testIntegration(opts, runCode);
```

Comment:
Sure, I have more code .. uploaded just to make it clean for the first milestone.